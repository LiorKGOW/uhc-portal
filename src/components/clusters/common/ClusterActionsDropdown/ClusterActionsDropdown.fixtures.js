import clusterStates from '../clusterStates';

const cluster = {
  id: 1,
  name: 'test-cluster',
  display_name: 'test-cluster',
  state: clusterStates.READY,
  console: { url: 'www.testuhc.com' },
  managed: true,
  ccs: {
    enabled: false,
  },
  canEdit: true,
  canDelete: true,
  status: {
    state: clusterStates.READY,
    dns_ready: true,
    configuration_mode: 'full',
  },
  subscription: {
    id: 'subscription-id',
  },
};

const osdTrialCluster = {
  product: {
    kind: 'ProductLink',
    // value after normalization
    id: 'OSDTrial',
    href: '/api/clusters_mgmt/v1/products/osdtrial?trial=osd',
  },
  ...cluster,
};

const props = {
  showConsoleButton: true,
  canSubscribeOCP: false,
  canHibernateCluster: true,
  canTransferClusterOwnership: false,
  openModal: jest.fn(),
  toggleSubscriptionReleased: jest.fn(),
  refreshFunc: jest.fn(),
};

const managedReadyProps = {
  cluster: { ...cluster },
  ...props,
};

const managedReadyOsdTrialProps = {
  cluster: { ...osdTrialCluster },
  ...props,
};

const hibernateClusterModalData = {
  clusterID: cluster.id,
  clusterName: cluster.name,
  subscriptionID: cluster.subscription.id,
};

const deleteModalData = {
  clusterID: cluster.id,
  clusterName: cluster.name,
};

const clusterHibernatingProps = {
  cluster: {
    ...cluster,
    state: clusterStates.HIBERNATING,
    status: {
      ...cluster.status,
      state: clusterStates.HIBERNATING,
    },
  },
  ...props,
};

const clusterUninstallingProps = {
  cluster: {
    ...cluster,
    state: clusterStates.UNINSTALLING,
    status: {
      ...cluster.status,
      state: clusterStates.UNINSTALLING,
    },
  },
  ...props,
};

const clusterNotReadyProps = {
  cluster: {
    ...cluster,
    console: undefined,
    state: clusterStates.ERROR,
    status: {
      state: clusterStates.ERROR,
      dns_ready: false,
      provision_error_message: 'some message',
      provision_error_code: 'some code',
    },
  },
  ...props,
};

const clusterReadOnlyProps = {
  cluster: {
    ...cluster,
    status: {
      ...cluster.status,
      configuration_mode: 'read_only',
    },
  },
  ...props,
};

const selfManagedProps = {
  cluster: {
    state: clusterStates.READY,
    managed: false,
    ccs: {
      enabled: false,
    },
    canEdit: true,
    canDelete: true,
  },
  subscription: {
    items: [
      {
        id: 42,
        support_level: 'Standard',
        plan: {
          id: 'OCP',
        },
      },
    ],
  },
  organization: {
    ebs_account_id: '123456',
  },
  ...props,
};

const organizationClusterProps = {
  cluster: { ...cluster, canEdit: false, canDelete: false },
  ...props,
};

export {
  managedReadyProps,
  managedReadyOsdTrialProps,
  deleteModalData,
  clusterHibernatingProps,
  clusterUninstallingProps,
  clusterNotReadyProps,
  clusterReadOnlyProps,
  selfManagedProps,
  cluster,
  osdTrialCluster,
  organizationClusterProps,
  hibernateClusterModalData,
};
