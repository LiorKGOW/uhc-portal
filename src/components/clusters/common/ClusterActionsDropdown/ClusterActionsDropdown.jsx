import React from 'react';
import PropTypes from 'prop-types';
import { MenuItem } from 'patternfly-react';
import clusterStates from '../clusterStates';

const ClusterActionsDropdown = (props) => {
  const {
    cluster,
    showConsoleButton,
    openModal,
    openEditClusterDialog,
  } = props;

  const uninstallingMessage = 'The cluster is being uninstalled';
  const consoleDisabledMessage = 'Admin console is not yet available for this cluster';
  const selfManagedEditMessage = 'Self managed cluster cannot be edited';
  const notReadyMessage = 'This cluster is not ready';

  const isClusterUninstalling = cluster.state === clusterStates.UNINSTALLING;
  const isClusterReady = cluster.state === clusterStates.READY;

  const isUninstallingProps = isClusterUninstalling
    ? { disabled: true, title: uninstallingMessage } : {};

  const getAdminConosleProps = () => {
    const consoleURL = cluster.console ? cluster.console.url : false;

    const adminConsoleEnabled = {
      href: consoleURL,
      target: '_blank',
      rel: 'noreferrer',
    };

    const adminConsoleDisabled = {
      disabled: true,
      title: isClusterUninstalling ? uninstallingMessage : consoleDisabledMessage,
    };

    return consoleURL && !isClusterUninstalling ? adminConsoleEnabled : adminConsoleDisabled;
  };

  const getEditClusterProps = () => {
    const selfManagedEditProps = {
      disabled: true,
      title: isClusterUninstalling ? uninstallingMessage : selfManagedEditMessage,
    };

    const managedEditProps = { onClick: () => openEditClusterDialog(cluster) };

    const disabledManagedEditProps = {
      disabled: true,
      title: isClusterUninstalling ? uninstallingMessage : notReadyMessage,
    };

    if (!cluster.managed) {
      return selfManagedEditProps;
    }
    return isClusterReady ? managedEditProps : disabledManagedEditProps;
  };

  const getEditDisplayNameProps = () => {
    const editDisplayNameProps = {
      onClick: () => openModal('edit-display-name', cluster),
    };

    return isClusterUninstalling ? isUninstallingProps : editDisplayNameProps;
  };

  const getDeleteItemProps = () => {
    const deleteModalData = {
      clusterID: cluster.id,
      clusterName: cluster.name,
      managed: cluster.managed,
    };

    return isClusterUninstalling
      ? isUninstallingProps : { onClick: () => openModal('delete-cluster', deleteModalData) };
  };

  const adminConsoleItemProps = getAdminConosleProps();
  const editClusterItemProps = getEditClusterProps();
  const editDisplayNameItemProps = getEditDisplayNameProps();
  const deleteClusterItemProps = getDeleteItemProps();

  return (
    <React.Fragment>
      {showConsoleButton && <MenuItem {...adminConsoleItemProps}>Launch Admin Console</MenuItem>}
      <MenuItem {...editDisplayNameItemProps}>Edit Display Name</MenuItem>
      <MenuItem {...editClusterItemProps}>Edit Cluster</MenuItem>
      <MenuItem {...deleteClusterItemProps}>Delete Cluster</MenuItem>
    </React.Fragment>
  );
};

ClusterActionsDropdown.propTypes = {
  cluster: PropTypes.object.isRequired,
  showConsoleButton: PropTypes.bool.isRequired,
  openModal: PropTypes.func.isRequired,
  openEditClusterDialog: PropTypes.func.isRequired,
};

export default ClusterActionsDropdown;
