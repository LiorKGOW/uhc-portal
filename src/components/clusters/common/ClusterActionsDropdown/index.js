import { connect } from 'react-redux';

import { openModal } from '../../../common/Modal/ModalActions';

import ClusterActionsDropdown from './ClusterActionsDropdown';

const mapDispatchToProps = {
  openModal,
};

export default connect(null, mapDispatchToProps)(ClusterActionsDropdown);
