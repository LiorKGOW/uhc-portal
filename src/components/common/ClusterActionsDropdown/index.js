
import { connect } from 'react-redux';
import { openModal } from '../Modal/ModalActions';
import ClusterActionsDropdown from './ClusterActionsDropdown';

const mapDispatchToProps = {
  openModal,
};

export default connect(null, mapDispatchToProps)(ClusterActionsDropdown);
