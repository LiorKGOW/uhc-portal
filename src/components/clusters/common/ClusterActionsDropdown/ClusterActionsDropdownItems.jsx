import get from 'lodash/get';
import React from 'react';
import { DropdownItem } from '@patternfly/react-core';
import { isRestrictedEnv } from '~/restrictedEnv';
import clusterStates, { isHibernating } from '../clusterStates';
import { subscriptionStatuses, normalizedProducts } from '../../../../common/subscriptionTypes';
import getClusterName from '../../../../common/getClusterName';
import modals from '../../../common/Modal/modals';
import { isAssistedInstallCluster } from '../../../../common/isAssistedInstallerCluster';
import { isHypershiftCluster } from '../../ClusterDetails/clusterDetailsHelper';

/**
 * Helper using reason message why it's disabled as source-of-truth
 * for whether it should be disabled.
 * This allows easy chaining `disableIfTooltip(reason1 || reason2 || ...)`.
 *
 * @param tooltip - message to show.  If truthy, also returns `isAriaDisabled: true` prop.
 * @param [propsIfEnabled] - return value if `tooltip` was falsy (default {}).
 */
const disableIfTooltip = (tooltip, propsIfEnabled = {}) =>
  // isDisabled blocks mouse events, so tooltip doesn't show on hover.
  // isAriaDisabled solved this, https://github.com/patternfly/patternfly-react/pull/6038.
  tooltip ? { isAriaDisabled: true, tooltip, tooltipProps: { position: 'left' } } : propsIfEnabled;

/**
 * This function is used by PF tables to determine which dropdown items are displayed
 * on each row of the table. It returns a list of objects, containing props for DropdownItem
 * PF table renders automatically.
 * @param {*} cluster             The cluster object corresponding to the current row
 * @param {*} showConsoleButton   true if 'Open Console' button should be displayed
 * @param {*} openModal           Action to open modal
 */
function actionResolver(
  cluster,
  showConsoleButton,
  openModal,
  canSubscribeOCP,
  canTransferClusterOwnership,
  canHibernateCluster,
  toggleSubscriptionReleased,
  refreshFunc,
  inClusterList,
) {
  const baseProps = {
    component: 'button',
  };
  const isClusterUninstalling = cluster.state === clusterStates.UNINSTALLING;
  const uninstallingMessage = isClusterUninstalling && (
    <span>The cluster is being uninstalled</span>
  );
  const isClusterReady = cluster.state === clusterStates.READY;
  // Superset of more specific uninstallingMessage.
  const notReadyMessage = !isClusterReady && <span>This cluster is not ready</span>;
  const isClusterInHibernatingProcess = isHibernating(cluster.state);
  const hibernatingMessage =
    isClusterInHibernatingProcess &&
    (cluster.state === clusterStates.RESUMING ? (
      <>This cluster is resuming; wait for it to be ready in order to perform actions</>
    ) : (
      <>This cluster is hibernating; resume cluster in order to perform actions</>
    ));

  const isClusterHibernatingOrPoweringDown =
    cluster.state === clusterStates.HIBERNATING || cluster.state === clusterStates.POWERING_DOWN;
  const isClusterPoweringDown = cluster.state === clusterStates.POWERING_DOWN;
  const poweringDownMessage = isClusterPoweringDown && (
    <span>
      This cluster is powering down; you will be able to resume after it reaches hibernating state.
    </span>
  );

  const isReadOnly = cluster?.status?.configuration_mode === 'read_only';
  const readOnlyMessage = isReadOnly && (
    <span>This operation is not available during maintenance</span>
  );

  const consoleURL = get(cluster, 'console.url', false);
  const consoleDisabledMessage = !consoleURL && (
    <span>Admin console is not yet available for this cluster</span>
  );

  const getKey = (item) => `${cluster.id}.menu.${item}`;
  const clusterName = getClusterName(cluster);
  const isProductOSDTrial = cluster.product && cluster.product.id === normalizedProducts.OSDTrial;

  const getAdminConsoleProps = () => ({
    ...baseProps,
    title: 'Open console',
    key: getKey('adminconsole'),
    ...disableIfTooltip(uninstallingMessage || hibernatingMessage || consoleDisabledMessage, {
      component: 'a',
      href: consoleURL,
      target: '_blank',
      rel: 'noopener noreferrer',
    }),
  });

  const getHibernateClusterProps = () => {
    const hibernateClusterBaseProps = {
      ...baseProps,
      key: getKey('hibernatecluster'),
    };
    const modalData = {
      clusterID: cluster.id,
      clusterName,
      subscriptionID: cluster.subscription ? cluster.subscription.id : '',
      shouldDisplayClusterName: inClusterList,
    };
    const hibernateClusterProps = {
      ...hibernateClusterBaseProps,
      title: 'Hibernate cluster',
      ...disableIfTooltip(uninstallingMessage || readOnlyMessage || notReadyMessage, {
        onClick: () => openModal(modals.HIBERNATE_CLUSTER, modalData),
      }),
    };
    const resumeHibernatingClusterProps = {
      ...hibernateClusterBaseProps,
      title: 'Resume from Hibernation',
      ...disableIfTooltip(poweringDownMessage || readOnlyMessage, {
        onClick: () => openModal(modals.RESUME_CLUSTER, modalData),
      }),
    };

    if (isClusterHibernatingOrPoweringDown) {
      return resumeHibernatingClusterProps;
    }
    return hibernateClusterProps;
  };

  const getScaleClusterProps = () => ({
    ...baseProps,
    title: 'Edit load balancers and persistent storage',
    key: getKey('scalecluster'),
    ...disableIfTooltip(
      uninstallingMessage || readOnlyMessage || hibernatingMessage || notReadyMessage,
      {
        onClick: () =>
          openModal(modals.SCALE_CLUSTER, { ...cluster, shouldDisplayClusterName: inClusterList }),
      },
    ),
  });

  const getEditNodeCountProps = () => ({
    ...baseProps,
    title: 'Edit node count',
    key: getKey('editnodecount'),
    ...disableIfTooltip(
      uninstallingMessage || readOnlyMessage || hibernatingMessage || notReadyMessage,
      {
        onClick: () =>
          openModal(modals.EDIT_NODE_COUNT, {
            cluster,
            isDefaultMachinePool: !isHypershiftCluster(cluster),
            shouldDisplayClusterName: inClusterList,
            isHypershiftCluster: isHypershiftCluster(cluster),
            clearMachineOrNodePoolsOnExit: true,
          }),
      },
    ),
  });

  const getEditDisplayNameProps = () => ({
    ...baseProps,
    title: 'Edit display name',
    key: getKey('editdisplayname'),
    ...disableIfTooltip(uninstallingMessage, {
      onClick: () =>
        openModal(modals.EDIT_DISPLAY_NAME, {
          ...cluster,
          shouldDisplayClusterName: inClusterList,
        }),
    }),
  });
  const getArchiveClusterProps = () => {
    const baseArchiveProps = {
      ...baseProps,
      title: 'Archive cluster',
      key: getKey('archivecluster'),
    };
    const archiveModalData = {
      subscriptionID: cluster.subscription ? cluster.subscription.id : '',
      name: clusterName,
    };
    return {
      ...baseArchiveProps,
      ...disableIfTooltip(readOnlyMessage, {
        onClick: () =>
          openModal(modals.ARCHIVE_CLUSTER, {
            ...archiveModalData,
            shouldDisplayClusterName: inClusterList,
          }),
      }),
    };
  };

  const getUnarchiveClusterProps = () => {
    const baseArchiveProps = {
      ...baseProps,
      title: 'Unarchive cluster',
      key: getKey('unarchivecluster'),
    };
    const unarchiveModalData = {
      subscriptionID: cluster.subscription ? cluster.subscription.id : '',
      name: clusterName,
    };
    return {
      ...baseArchiveProps,
      ...disableIfTooltip(readOnlyMessage, {
        onClick: () =>
          openModal(modals.UNARCHIVE_CLUSTER, {
            ...unarchiveModalData,
            shouldDisplayClusterName: inClusterList,
          }),
      }),
    };
  };

  const getEditConsoleURLProps = () => ({
    ...baseProps,
    key: getKey('editconsoleurl'),
    ...disableIfTooltip(uninstallingMessage, {
      title: consoleURL ? 'Edit console URL' : 'Add console URL',
      onClick: () =>
        openModal(modals.EDIT_CONSOLE_URL, { ...cluster, shouldDisplayClusterName: inClusterList }),
    }),
  });

  const getDeleteItemProps = () => ({
    ...baseProps,
    title: 'Delete cluster',
    key: getKey('deletecluster'),
    ...disableIfTooltip(uninstallingMessage || readOnlyMessage || hibernatingMessage, {
      onClick: () =>
        openModal(modals.DELETE_CLUSTER, {
          clusterID: cluster.id,
          clusterName,
          shouldDisplayClusterName: inClusterList,
        }),
    }),
  });

  const getEditSubscriptionSettingsProps = () => {
    const editSubscriptionSettingsProps = {
      ...baseProps,
      title: 'Edit subscription settings',
      key: getKey('editsubscriptionsettings'),
      onClick: () =>
        openModal(modals.EDIT_SUBSCRIPTION_SETTINGS, {
          subscription: cluster.subscription,
          shouldDisplayClusterName: inClusterList,
        }),
    };
    return editSubscriptionSettingsProps;
  };

  const getTransferClusterOwnershipProps = () => {
    const isReleased = get(cluster, 'subscription.released', false);
    const title = isReleased ? 'Cancel ownership transfer' : 'Transfer cluster ownership';
    const transferClusterOwnershipProps = {
      ...baseProps,
      title,
      key: getKey('transferclusterownership'),
      onClick: () => {
        if (isReleased) {
          toggleSubscriptionReleased(get(cluster, 'subscription.id'), false);
          refreshFunc();
        } else {
          openModal(modals.TRANSFER_CLUSTER_OWNERSHIP, {
            subscription: cluster.subscription,
            shouldDisplayClusterName: inClusterList,
          });
        }
      },
    };
    return transferClusterOwnershipProps;
  };

  const getUpgradeTrialClusterProps = () => {
    const upgradeTrialClusterData = {
      clusterID: cluster.id,
      cluster,
    };
    const upgradeTrialClusterProps = {
      ...baseProps,
      title: 'Upgrade cluster from Trial',
      key: getKey('upgradetrialcluster'),
    };
    return {
      ...upgradeTrialClusterProps,
      ...disableIfTooltip(readOnlyMessage, {
        onClick: () =>
          openModal(modals.UPGRADE_TRIAL_CLUSTER, {
            ...upgradeTrialClusterData,
            shouldDisplayClusterName: inClusterList,
          }),
      }),
    };
  };

  const adminConsoleItemProps = getAdminConsoleProps();
  const scaleClusterItemProps = getScaleClusterProps();
  const editNodeCountItemProps = getEditNodeCountProps();
  const editDisplayNameItemProps = getEditDisplayNameProps();
  const editConsoleURLItemProps = getEditConsoleURLProps();
  const deleteClusterItemProps = getDeleteItemProps();
  const archiveClusterItemProps = getArchiveClusterProps();
  const unarchiveClusterItemProps = getUnarchiveClusterProps();
  const editSubscriptionSettingsProps = getEditSubscriptionSettingsProps();
  const transferClusterOwnershipProps = getTransferClusterOwnershipProps();
  const upgradeTrialClusterProps = getUpgradeTrialClusterProps();
  const hibernateClusterProps = getHibernateClusterProps();

  const showDelete = cluster.canDelete && cluster.managed;
  const showScale = cluster.canEdit && cluster.managed && !cluster.ccs?.enabled;
  const showHibernateCluster =
    cluster.canEdit &&
    cluster.managed &&
    canHibernateCluster &&
    !isProductOSDTrial &&
    !isHypershiftCluster(cluster);
  const showEditNodeCount = cluster.canEdit && cluster.managed;
  const isArchived = get(cluster, 'subscription.status', false) === subscriptionStatuses.ARCHIVED;
  const showArchive = cluster.canEdit && !cluster.managed && cluster.subscription && !isArchived;
  const showUnarchive = cluster.canEdit && !cluster.managed && cluster.subscription && isArchived;
  const showEditURL =
    !cluster.managed &&
    cluster.canEdit &&
    (showConsoleButton || consoleURL) &&
    !isAssistedInstallCluster(cluster);
  const product = get(cluster, 'subscription.plan.type', '');
  const showEditSubscriptionSettings =
    product === normalizedProducts.OCP && cluster.canEdit && canSubscribeOCP;
  const isAllowedProducts = [normalizedProducts.OCP, normalizedProducts.ARO].includes(product);
  const showTransferClusterOwnership =
    cluster.canEdit &&
    canTransferClusterOwnership &&
    isAllowedProducts &&
    get(cluster, 'subscription.status') !== subscriptionStatuses.ARCHIVED;
  const showUpgradeTrialCluster = isClusterReady && cluster.canEdit && isProductOSDTrial;

  return [
    showConsoleButton && adminConsoleItemProps,
    cluster.canEdit && editDisplayNameItemProps,
    showEditURL && editConsoleURLItemProps,
    showScale && scaleClusterItemProps,
    showEditNodeCount && editNodeCountItemProps,
    showHibernateCluster && hibernateClusterProps,
    showUpgradeTrialCluster && upgradeTrialClusterProps,
    showDelete && deleteClusterItemProps,
    showArchive && archiveClusterItemProps,
    showUnarchive && unarchiveClusterItemProps,
    showEditSubscriptionSettings && editSubscriptionSettingsProps,
    showTransferClusterOwnership && transferClusterOwnershipProps,
  ].filter(Boolean);
}

function dropDownItems({
  cluster,
  showConsoleButton,
  openModal,
  canSubscribeOCP,
  canTransferClusterOwnership,
  canHibernateCluster,
  toggleSubscriptionReleased,
  refreshFunc,
  inClusterList,
}) {
  const actions = actionResolver(
    cluster,
    showConsoleButton,
    openModal,
    canSubscribeOCP,
    canTransferClusterOwnership,
    canHibernateCluster,
    toggleSubscriptionReleased,
    refreshFunc,
    inClusterList,
  );
  const menuItems = actions.map((action) => (
    <DropdownItem {...action}>{action.title}</DropdownItem>
  ));
  return menuItems;
}

export { actionResolver, dropDownItems };
