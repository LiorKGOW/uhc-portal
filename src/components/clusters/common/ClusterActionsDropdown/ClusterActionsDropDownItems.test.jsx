import React from 'react';
import { shallow } from 'enzyme';
import { DropdownItem } from '@patternfly/react-core';

import { dropDownItems } from './ClusterActionsDropdownItems';
import * as Fixtures from './ClusterActionsDropdown.fixtures';

function DropDownItemsRenderHelper(props) {
  return (
    <>
      {dropDownItems(props).map(item => item)}
    </>
  );
}

describe('Cluster Actions Dropdown Items', () => {
  describe('cluster with state ready and console url', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(<DropDownItemsRenderHelper {...Fixtures.managedReadyProps} />);
    });

    it('should render', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should open edit display name modal', () => {
      wrapper.find(DropdownItem).at(1).simulate('click');
      expect(Fixtures.managedReadyProps.openModal).toBeCalledWith('edit-display-name', Fixtures.cluster);
    });

    it('should open edit cluster modal', () => {
      wrapper.find(DropdownItem).at(2).simulate('click');
      expect(Fixtures.managedReadyProps.openModal).toBeCalledWith('edit-cluster', Fixtures.cluster);
    });

    it('should open delete modal', () => {
      wrapper.find(DropdownItem).at(3).simulate('click');
      expect(Fixtures.managedReadyProps.openModal).toBeCalledWith('delete-cluster', Fixtures.deleteModalData);
    });

    it('menu buttons should be enabled', () => {
      const launchConsoleDisabled = wrapper.find(DropdownItem).at(0).props().isDisabled;
      const editDisplayNameDisabled = wrapper.find(DropdownItem).at(1).props().isDisabled;
      const editDisabled = wrapper.find(DropdownItem).at(2).props().isDisabled;
      const deleteDisabled = wrapper.find(DropdownItem).at(3).props().isDisabled;
      expect(launchConsoleDisabled).toBeFalsy();
      expect(editDisplayNameDisabled).toBeFalsy();
      expect(editDisabled).toBeFalsy();
      expect(deleteDisabled).toBeFalsy();
    });
  });

  describe('cluster with state uninstalling', () => {
    const wrapper = shallow(<DropDownItemsRenderHelper {...Fixtures.clusterUninstallingProps} />);

    it('should render (uninstalling)', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('menu buttons should be disabled', () => {
      const launchConsoleDisabled = wrapper.find(DropdownItem).at(0).props().isDisabled;
      const editDisplayNameDisabled = wrapper.find(DropdownItem).at(1).props().isDisabled;
      const editDisabled = wrapper.find(DropdownItem).at(2).props().isDisabled;
      const deleteDisabled = wrapper.find(DropdownItem).at(3).props().isDisabled;
      expect(launchConsoleDisabled).toEqual(true);
      expect(editDisplayNameDisabled).toEqual(true);
      expect(editDisabled).toEqual(true);
      expect(deleteDisabled).toEqual(true);
    });
  });

  describe('cluster with state not ready', () => {
    const wrapper = shallow(<DropDownItemsRenderHelper {...Fixtures.clusterNotReadyProps} />);

    it('should render (not ready)', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('disable launch console & edit cluster, enable edit display name and delete cluster', () => {
      const launchConsoleDisabled = wrapper.find(DropdownItem).at(0).props().isDisabled;
      const editDisplayNameDisabled = wrapper.find(DropdownItem).at(1).props().isDisabled;
      const editDisabled = wrapper.find(DropdownItem).at(2).props().isDisabled;
      const deleteDisabled = wrapper.find(DropdownItem).at(3).props().isDisabled;
      expect(launchConsoleDisabled).toEqual(true);
      expect(editDisplayNameDisabled).toBeFalsy();
      expect(editDisabled).toEqual(true);
      expect(deleteDisabled).toBeFalsy();
    });
  });

  describe('self managed cluster', () => {
    const wrapper = shallow(<DropDownItemsRenderHelper {...Fixtures.selfManagedProps} />);

    it('should render (self managed)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('admin console url does not exist', () => {
    const wrapper = shallow(<DropDownItemsRenderHelper {...Fixtures.selfManagedProps} />);

    it('should render (no console)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('read only cluster', () => {
    const wrapper = shallow(<DropDownItemsRenderHelper {...Fixtures.organizationClusterProps} />);
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
