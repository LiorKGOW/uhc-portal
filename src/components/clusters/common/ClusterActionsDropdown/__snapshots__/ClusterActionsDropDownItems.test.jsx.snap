// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cluster Actions Dropdown Items admin console url does not exist should render (no console) 1`] = `
<Fragment>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={true}
    isHovered={false}
    key="undefined.menu.adminconsole"
    tooltip={
      <span>
        Admin console is not yet available for this cluster
      </span>
    }
    tooltipProps={Object {}}
  >
    Launch Admin Console
  </DropdownItem>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={false}
    isHovered={false}
    key="undefined.menu.editdisplayname"
    onClick={[Function]}
    tooltip={null}
    tooltipProps={Object {}}
  >
    Edit Display Name
  </DropdownItem>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={true}
    isHovered={false}
    key="undefined.menu.editcluster"
    tooltip={
      <span>
        Self managed cluster cannot be edited
      </span>
    }
    tooltipProps={Object {}}
  >
    Scale Cluster
  </DropdownItem>
</Fragment>
`;

exports[`Cluster Actions Dropdown Items cluster with state not ready should render (not ready) 1`] = `
<Fragment>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={true}
    isHovered={false}
    key="undefined.menu.adminconsole"
    tooltip={
      <span>
        Admin console is not yet available for this cluster
      </span>
    }
    tooltipProps={Object {}}
  >
    Launch Admin Console
  </DropdownItem>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={false}
    isHovered={false}
    key="undefined.menu.editdisplayname"
    onClick={[Function]}
    tooltip={null}
    tooltipProps={Object {}}
  >
    Edit Display Name
  </DropdownItem>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={true}
    isHovered={false}
    key="undefined.menu.editcluster"
    tooltip={
      <span>
        This cluster is not ready
      </span>
    }
    tooltipProps={Object {}}
  >
    Scale Cluster
  </DropdownItem>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={false}
    isHovered={false}
    key="undefined.menu.deletecluster"
    onClick={[Function]}
    tooltip={null}
    tooltipProps={Object {}}
  >
    Delete Cluster
  </DropdownItem>
</Fragment>
`;

exports[`Cluster Actions Dropdown Items cluster with state ready and console url should render 1`] = `
<Fragment>
  <DropdownItem
    className=""
    component="button"
    href="www.testuhc.com"
    isDisabled={false}
    isHovered={false}
    key="1.menu.adminconsole"
    rel="noreferrer"
    target="_blank"
    tooltip={null}
    tooltipProps={Object {}}
  >
    Launch Admin Console
  </DropdownItem>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={false}
    isHovered={false}
    key="1.menu.editdisplayname"
    onClick={[Function]}
    tooltip={null}
    tooltipProps={Object {}}
  >
    Edit Display Name
  </DropdownItem>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={false}
    isHovered={false}
    key="1.menu.editcluster"
    onClick={[Function]}
    tooltip={null}
    tooltipProps={Object {}}
  >
    Scale Cluster
  </DropdownItem>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={false}
    isHovered={false}
    key="1.menu.deletecluster"
    onClick={[Function]}
    tooltip={null}
    tooltipProps={Object {}}
  >
    Delete Cluster
  </DropdownItem>
</Fragment>
`;

exports[`Cluster Actions Dropdown Items cluster with state uninstalling should render (uninstalling) 1`] = `
<Fragment>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={true}
    isHovered={false}
    key="undefined.menu.adminconsole"
    tooltip={
      <span>
        The cluster is being uninstalled
      </span>
    }
    tooltipProps={Object {}}
  >
    Launch Admin Console
  </DropdownItem>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={true}
    isHovered={false}
    key="undefined.menu.editdisplayname"
    tooltip={
      <span>
        The cluster is being uninstalled
      </span>
    }
    tooltipProps={Object {}}
  >
    Edit Display Name
  </DropdownItem>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={true}
    isHovered={false}
    key="undefined.menu.editcluster"
    tooltip={
      <span>
        The cluster is being uninstalled
      </span>
    }
    tooltipProps={Object {}}
  >
    Scale Cluster
  </DropdownItem>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={true}
    isHovered={false}
    key="undefined.menu.deletecluster"
    tooltip={
      <span>
        The cluster is being uninstalled
      </span>
    }
    tooltipProps={Object {}}
  >
    Delete Cluster
  </DropdownItem>
</Fragment>
`;

exports[`Cluster Actions Dropdown Items read only cluster should render correctly 1`] = `
<Fragment>
  <DropdownItem
    className=""
    component="button"
    href="www.testuhc.com"
    isDisabled={false}
    isHovered={false}
    key="1.menu.adminconsole"
    rel="noreferrer"
    target="_blank"
    tooltip={null}
    tooltipProps={Object {}}
  >
    Launch Admin Console
  </DropdownItem>
</Fragment>
`;

exports[`Cluster Actions Dropdown Items self managed cluster should render (self managed) 1`] = `
<Fragment>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={true}
    isHovered={false}
    key="undefined.menu.adminconsole"
    tooltip={
      <span>
        Admin console is not yet available for this cluster
      </span>
    }
    tooltipProps={Object {}}
  >
    Launch Admin Console
  </DropdownItem>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={false}
    isHovered={false}
    key="undefined.menu.editdisplayname"
    onClick={[Function]}
    tooltip={null}
    tooltipProps={Object {}}
  >
    Edit Display Name
  </DropdownItem>
  <DropdownItem
    className=""
    component="button"
    href=""
    isDisabled={true}
    isHovered={false}
    key="undefined.menu.editcluster"
    tooltip={
      <span>
        Self managed cluster cannot be edited
      </span>
    }
    tooltipProps={Object {}}
  >
    Scale Cluster
  </DropdownItem>
</Fragment>
`;
